mapper/dm_agent/src/monitor.rs

```diff
@@ -1,13 +1,23 @@
 use mqtt_client::{Client, MqttClient};
 use std::sync::Arc;
 use thin_edge_json::group::MeasurementGrouper;
+use time::WallClock;
 use tracing::{instrument, log::error};
 
-use crate::batcher::{DeviceMonitorError, MessageBatchPublisher, MessageBatcher};
+use crate::{
+    batcher::{MessageBatchPublisher, MessageBatcher},
+    error::DeviceMonitorError,
+};
 
 const DEFAULT_HOST: &str = "localhost";
```

Comment:
I generally like to avoid declaring top level constants and using them throughout the code base. That's bad for testing.

Instead I would add a `DmAgentConfig`:

```rust
struct DmAgentConfig {
   port: u16,
  stat_collection_window: u64,
  source_topic: Topic,
  target_topic: Topic,
}

impl Default for DmAgentConfig {
  fn default() -> Self {
    Self {
       // ... put the defaults here
    }
  }
```

I'd use `unwrap`s for converting the `SOURCE_TOPIC` into a `Topic` and create the `DmAgentConfig` early on. Whether you `panic` at startup (due to an invalid default topic name written by a programmer) or just abort due to an `Err`... I prefer to panic (for as long as you are constructing that default config within your `main` function).